src/arithmetic/numeric_funcs/numeric_funcs.c

```diff
@@ -120,6 +121,30 @@ SIValue AR_SIGN(SIValue *argv, int argc) {
 	return SI_LongVal(sign);
 }
 
+SIValue AR_TOINTEGER(SIValue *argv, int argc) {
+	SIValue arg = argv[0];
+	char *sEnd = NULL;
+
+	switch(SI_TYPE(arg)) {
+	case T_INT64:
+		return arg;
+	case T_DOUBLE:
+		// Remove floating point.
+		return SI_LongVal(floor(arg.doubleval));
+	case T_STRING:
+		errno = 0;
+		double parsedval = strtod(arg.stringval, &sEnd);
+		/* The input was not a complete number or represented a number that
+		 * cannot be represented as a double.
+		 * Create a string SIValue. */
+		if(sEnd[0] != '\0' || errno == ERANGE) return SI_NullVal();
+		// Remove floating point.
+		return SI_LongVal(floor(parsedval));
+	default:
```

Comment:
This function should also handle NULL values, with NULL inputs returning null SIValues. This is important for cases like operating on a property that may not exist on all inputs.

Reply:
👍